### PR TITLE
fix 2 vanilla issues

### PR DESCRIPTION
**Pull request by Machine-Maker**
Fix sheeps eating grass outside ticking chunks
Fix itemframe drops dropping in correct location

Replacement for https://github.com/PaperMC/Paper/pull/8420 and https://github.com/PaperMC/Paper/pull/8424

### CODE REVIEW

**Review by Owen1212055 — APPROVED** [review]

File: patches/server/0899-Fix-a-bunch-of-vanilla-bugs.patch [hunk 1]
```diff
@@ -44,6 +44,14 @@ https://bugs.mojang.com/browse/MC-258535
   by: Jake Potrebic <jake.m.potrebic@gmail.com>
   Fixes certain explosion damage not scaling with difficulty
 
+https://bugs.mojang.com/browse/MC-210802
+  by: BillyGalbreath <Blake.Galbreath@Gmail.com>
```
Comment (Owen1212055) [inline patches/server/0899-Fix-a-bunch-of-vanilla-bugs.patch]: Also, diff capitalization on emails?

Comment (Machine-Maker) [reply patches/server/0899-Fix-a-bunch-of-vanilla-bugs.patch]: Just copied them right from the original patch files.

**Comment by lynxplay** [comment]
For documentation purposes, `shouldTickBlocksAt` does not work here, spawn chunks, even if not random ticked, remain at FullStatus ENTITY_TICKING and hence `shouldTickBlocksAt` returns true for blocks that are not random ticked.

**Comment by Machine-Maker** [comment]
Ok, changed the check in EatBlockGoal to check if any player was near enough for spawning, the same check used for random ticking in ServerChunkCache

**Pull request merged**